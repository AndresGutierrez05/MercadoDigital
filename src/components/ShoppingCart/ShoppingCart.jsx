import React, { useState } from "react";
import { useSelector } from 'react-redux';
import Quantity from "../Quantity/Quantity";
import "./ShoppingCart.scss";

export default function ShoppingCart({ propCart }){
    const [ showShoppingCart, setShowShoppingCart ] = useState(false);
    const reduxCart = useSelector((state) => state.productosReducer.carrito) ?? [];
    const cart = propCart || reduxCart;
    const totalPrice = cart.map(product => product.Precio * product.cantidad).reduce((a, b) => a + b, 0) ?? 0;

    const handleShowShoppingCart = (show) => {
        setShowShoppingCart(show != null ? show : !showShoppingCart);
    }

    return(
        <div className="cart-container">
            <div className="cart-container-right">
                <div className="cart-button" onClick={cart.length > 0 ? () => handleShowShoppingCart(!showShoppingCart) : null}  onMouseEnter={cart.length > 0 ? handleShowShoppingCart : null}>
                    <div className="cart-button-img">
                        <img src="/icons/cart.png" alt="carrito"></img>
                    </div>
                    <div>
                        <p className="totalPrice">${totalPrice}</p>
                    </div>
                </div>
                <div className="cart-summary" style={{ display : cart.length > 0 && showShoppingCart ? "inline" : "none"}} onMouseLeave={() => handleShowShoppingCart(false)}>
                    <h2>Carrito</h2>
                    <hr/>
                    <div className="cart-summary-products">
                        {cart.map((product, i) =>
                            <div key={i}>
                                <div className="cart-summary-product">
                                    <Quantity quantity={product.cantidad}/>
                                    <img src={product.RutaImagen} alt={product.Nombre}></img>
                                </div>
                                <hr/>
                            </div>
                        )}
                    </div>
                    <div className="cart-total">
                        <div>
                            <form action="https://checkout.wompi.co/p/" method="GET">
                                <input type="hidden" name="public-key" value="pub_test_X0zDA9xoKdePzhd8a0x9HAez7HgGO2fH" />
                                <input type="hidden" name="currency" value="COP" />
                                <input type="hidden" name="amount-in-cents" value={`${totalPrice}00`} />
                                <input type="hidden" name="reference" value="4XMPGKWWPKWQ" />
                                <input type="hidden" name="signature-integrity" value="37c8407747e595535433ef8f6a811d853cd943046624a0ec04662b17bbf33bf5" />
                                <button type="submit">Paga con <strong>Wompi</strong></button>
                            </form>
                        </div>
                        <h3>Total : ${totalPrice}</h3>
                    </div>
                </div>
            </div>
        </div>
    )
} 