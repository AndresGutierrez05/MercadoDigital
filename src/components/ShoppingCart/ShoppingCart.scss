@use "sass:map";
@use "../../mixins.scss";
@use "../../commonstyles.scss" as common;

.cart-container{
    padding-top: 20px;
    width: 100%;
    display: flex;
    justify-content: flex-end;

    @media screen and (max-width: common.$medium) {
        padding-top: 5px;
    }

    .cart-container-right{
        .cart-button{
            display: flex;
            color: aliceblue;
            padding: 5px;
            margin-right: 30px;
            background-color: map.get(common.$theme-colors, "purple");
            border: 1px solid map.get(common.$theme-colors, "primary-blue");
            cursor: pointer;
            border-radius: 5px;

            @media screen and (max-width: common.$medium) {
                width: 100%;
                padding: 2px;
                margin-right: 5px;
            }
    
            div p {
                padding-left: 5px;
                font-weight: bold;
                font-size: larger;
            }
    
            .cart-button-img{
                align-self: center;
    
                img{
                    width: 40px;
                    height: 40px;

                    @media screen and (max-width: common.$small) {
                        width: 10px;
                        height: 10px;
                    }
                    @media screen and (max-width: common.$medium) {
                        width: 20px;
                        height: 20px;
                    }
                }
            }
        }
    
        .cart-summary{
            
            @include mixins.commonProductsContainer;
    
            z-index: 2;
            position: absolute;
            right: 0;
            padding: 20px;
            max-width: 400px;
            margin-right: 30px;
            width:40%;
            background-color: white;
            align-items: center;
            border-radius: 10px;
            border: 1px solid map.get(common.$theme-colors, "primary-blue");
            box-shadow: 0 2px 5px 2px map.get(common.$theme-colors, "primary-blue");

            @media screen and (max-width: common.$medium) {
                width:80%;
                margin-right: 0px;
            }

            img{
                max-width: 100px;
                border-radius: 5px;
            }
            
            .cart-summary-products{
                max-height: 600px;
                overflow-y: auto;

                @media screen and (max-width: common.$medium) {
                    max-height: 250px;
                }

                .cart-summary-product{
                    align-items: center;
                    display: flex;
                    justify-content: space-around;
                }
            }
    
            .cart-total{
                display: flex;
                justify-content: space-between;
                height: 100%;
                vertical-align: middle;
                
                
                h3{
                    color: map.get(common.$theme-colors, "purple");
                }

                div{
                    align-self: center;

                    button{
                        display: inline-block;
                        height: 40px;
                        line-height: 40px;
                        background-color: rgb(26, 69, 148);
                        border: 0px;
                        border-radius: 4px;
                        font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
                        font-weight: 400;
                        font-size: 14px;
                        padding: 0px 16px;
                        color: white;
                        cursor: pointer;
                        -webkit-font-smoothing: subpixel-antialiased;
                    }
                }
            }

        }
    }
}