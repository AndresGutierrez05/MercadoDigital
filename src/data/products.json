[   
    {
        "Id" : 1,
        "Nombre": "Atún en Lata",
        "Marca" : "Van Camps",
        "Unidad": "g",
        "Descripcion": "Atún en agua",
        "Complemento": "Incluye rallado, lomitos y esparcible",
        "NombrePorcion": "Porción",
        "TamañoPorcion": 80.000,
        "RutaImagen": "https://copservir.vtexassets.com/arquivos/ids/773477-800-auto?v=637951138022000000",
        "Precio" : 5400.0000
    },
    {
        "Id" : 2,
        "Nombre": "Leche Pasteurizada Entera",
        "Marca" : "Alqueria",
        "Unidad": "ml",
        "Descripcion": "Leche de vaca. Alimento básico y equilibrado con excelente densidad nutricional. Aporta proteínas de alto valor biológico, calcio, fósforo, vitaminas A, D y del complejo B.",
        "Complemento": "Cualquier tipo de leche líquida de vaca",
        "NombrePorcion": "Litro",
        "TamañoPorcion": 1300.000,
        "RutaImagen": "https://jumbocolombiaio.vtexassets.com/arquivos/ids/403045-1600-1600",
        "Precio" : 5900.0000
    },
    {
        "Id" : 3,
        "Nombre": "Azúcar Blanca",
        "Marca" : "Riopaila",
        "Unidad": "g",
        "Descripcion": "Azucar blanca refinada",
        "Complemento": "",
        "NombrePorcion": "Bolsa",
        "TamañoPorcion": 1000.000,
        "RutaImagen": "https://vixark.b-cdn.net/lp-i-i-g/azúcar-blanca-riopaila-1000g.jpg",
        "Precio" : 2700.0000
    },
    {
        "Id" : 4,
        "Nombre": "Aceite Vegetal",
        "Marca" : "Frescampo",
        "Unidad": "ml",
        "Descripcion": "Aceite Vegetal FRESCAMPO 3000 ml",
        "Complemento": "",
        "NombrePorcion": "Litro",
        "TamañoPorcion": 3000.000,
        "RutaImagen": "https://exitocol.vtexassets.com/arquivos/ids/13929565-500-auto?v=1753578000",
        "Precio" : 24980.0000
    },
    {
        "Id" : 5,
        "Nombre": "Jabon Avena Barra",
        "Marca" : "Protex",
        "Unidad": "g",
        "Descripcion": "Jabon Avena Barra X3 Unds PROTEX 330 gr",
        "Complemento": "",
        "NombrePorcion": "Barra",
        "TamañoPorcion": 330.000,
        "RutaImagen": "https://exitocol.vtexassets.com/arquivos/ids/14758134-1200-auto?v=637992541550430000&",
        "Precio" : 10900.0000
    },
    {
        "Id" : 6,
        "Nombre": "Atún en Lata",
        "Marca" : "Van Camps",
        "Unidad": "g",
        "Descripcion": "Atún en agua",
        "Complemento": "Incluye rallado, lomitos y esparcible",
        "NombrePorcion": "Porción",
        "TamañoPorcion": 80.000,
        "RutaImagen": "https://copservir.vtexassets.com/arquivos/ids/773477-800-auto?v=637951138022000000",
        "Precio" : 5400.0000
    },
    {
        "Id" : 7,
        "Nombre": "Leche Pasteurizada Entera",
        "Marca" : "Alqueria",
        "Unidad": "ml",
        "Descripcion": "Leche de vaca. Alimento básico y equilibrado con excelente densidad nutricional. Aporta proteínas de alto valor biológico, calcio, fósforo, vitaminas A, D y del complejo B.",
        "Complemento": "Cualquier tipo de leche líquida de vaca",
        "NombrePorcion": "Litro",
        "TamañoPorcion": 1000.000,
        "RutaImagen": "https://jumbocolombiaio.vtexassets.com/arquivos/ids/403045-1600-1600",
        "Precio" : 1900.0000
    },
    {
        "Id" : 8,
        "Nombre": "Azúcar Blanca",
        "Marca" : "Riopaila",
        "Unidad": "g",
        "Descripcion": "Azucar blanca refinada",
        "Complemento": "",
        "NombrePorcion": "Bolsa",
        "TamañoPorcion": 1000.000,
        "RutaImagen": "https://vixark.b-cdn.net/lp-i-i-g/azúcar-blanca-riopaila-1000g.jpg",
        "Precio" : 2700.0000
    },
    {
        "Id" : 9,
        "Nombre": "Aceite Vegetal",
        "Marca" : "Frescampo",
        "Unidad": "ml",
        "Descripcion": "Aceite Vegetal FRESCAMPO 3000 ml",
        "Complemento": "",
        "NombrePorcion": "Litro",
        "TamañoPorcion": 3000.000,
        "RutaImagen": "https://exitocol.vtexassets.com/arquivos/ids/13929565-500-auto?v=1753578000",
        "Precio" : 24980.0000
    },
    {
        "Id" : 10,
        "Nombre": "Jabon Avena Barra",
        "Marca" : "Protex",
        "Unidad": "g",
        "Descripcion": "Jabon Avena Barra X3 Unds PROTEX 330 gr",
        "Complemento": "",
        "NombrePorcion": "Barra",
        "TamañoPorcion": 330.000,
        "RutaImagen": "https://exitocol.vtexassets.com/arquivos/ids/14758134-1200-auto?v=637992541550430000&",
        "Precio" : 10900.0000
    },
    {
        "Id" : 11,
        "Nombre": "Atún en Lata",
        "Marca" : "Van Camps",
        "Unidad": "g",
        "Descripcion": "Atún en agua",
        "Complemento": "Incluye rallado, lomitos y esparcible",
        "NombrePorcion": "Porción",
        "TamañoPorcion": 80.000,
        "RutaImagen": "https://copservir.vtexassets.com/arquivos/ids/773477-800-auto?v=637951138022000000",
        "Precio" : 5400.0000
    },
    {
        "Id" : 12,
        "Nombre": "Leche Pasteurizada Entera",
        "Marca" : "Alqueria",
        "Unidad": "ml",
        "Descripcion": "Leche de vaca. Alimento básico y equilibrado con excelente densidad nutricional. Aporta proteínas de alto valor biológico, calcio, fósforo, vitaminas A, D y del complejo B.",
        "Complemento": "Cualquier tipo de leche líquida de vaca",
        "NombrePorcion": "Litro",
        "TamañoPorcion": 1000.000,
        "RutaImagen": "https://jumbocolombiaio.vtexassets.com/arquivos/ids/403045-1600-1600",
        "Precio" : 1900.0000
    },
    {
        "Id" : 13,
        "Nombre": "Azúcar Blanca",
        "Marca" : "Riopaila",
        "Unidad": "g",
        "Descripcion": "Azucar blanca refinada",
        "Complemento": "",
        "NombrePorcion": "Bolsa",
        "TamañoPorcion": 1000.000,
        "RutaImagen": "https://vixark.b-cdn.net/lp-i-i-g/azúcar-blanca-riopaila-1000g.jpg",
        "Precio" : 2700.0000
    },
    {
        "Id" : 14,
        "Nombre": "Aceite Vegetal",
        "Marca" : "Frescampo",
        "Unidad": "ml",
        "Descripcion": "Aceite Vegetal FRESCAMPO 3000 ml",
        "Complemento": "",
        "NombrePorcion": "Litro",
        "TamañoPorcion": 3000.000,
        "RutaImagen": "https://exitocol.vtexassets.com/arquivos/ids/13929565-500-auto?v=1753578000",
        "Precio" : 24980.0000
    },
    {
        "Id" : 15,
        "Nombre": "Jabon Avena Barra",
        "Marca" : "Protex",
        "Unidad": "g",
        "Descripcion": "Jabon Avena Barra X3 Unds PROTEX 330 gr",
        "Complemento": "",
        "NombrePorcion": "Barra",
        "TamañoPorcion": 330.000,
        "RutaImagen": "https://exitocol.vtexassets.com/arquivos/ids/14758134-1200-auto?v=637992541550430000&",
        "Precio" : 10900.0000
    }
]