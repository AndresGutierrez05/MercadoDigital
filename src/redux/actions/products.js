export const START_GET_PRODUCTS = 'START_GET_PRODUCTS';
export const SUCCESS_GET_PRODUCTS = 'SUCCESS_GET_PRODUCTS';
export const START_GET_PRODUCT = 'START_GET_PRODUCT';
export const SUCCESS_GET_PRODUCT = 'SUCCESS_GET_PRODUCT';
export const START_REMOVE_SELECTED_PRODUCT = 'START_REMOVE_SELECTED_PRODUCT';
export const SUCCESS_REMOVE_SELECTED_PRODUCT = 'SUCCESS_REMOVE_SELECTED_PRODUCT';
export const START_SET_CART = 'START_SET_CART';
export const SUCCESS_SET_CART = 'SUCCESS_SET_CART';
export const START_REMOVE_PRODUCT_CART = 'START_REMOVE_PRODUCT_CART';
export const SUCCESS_REMOVE_PRODUCT_CART = 'SUCCESS_REMOVE_PRODUCT_CART';
export const START_ADD_PRODUCT_QUANTITY = 'START_ADD_PRODUCT_QUANTITY';
export const SUCCESS_ADD_PRODUCT_QUANTITY = 'SUCCESS_ADD_PRODUCT_QUANTITY';
export const START_DISCOUNT_PRODUCT_QUANTITY = 'START_DISCOUNT_PRODUCT_QUANTITY';
export const SUCCESS_DISCOUNT_PRODUCT_QUANTITY = 'SUCCESS_DISCOUNT_PRODUCT_QUANTITY';

export const startGetProducts = payload => ({
    type: START_GET_PRODUCTS,
    ...payload
});

export const startProductSelected = payload => ({
    type: START_GET_PRODUCT,
    ...payload
});

export const startRemoveProductCart = payload => ({
    type: START_REMOVE_PRODUCT_CART,
    ...payload
});

export const startSetCart = payload => ({
    type: START_SET_CART,
    ...payload
});

export const startRemoveSelectedProduct = payload => ({
    type: START_REMOVE_SELECTED_PRODUCT,
    ...payload
});

export const startAddSelectedProduct = payload => ({
    type: START_ADD_PRODUCT_QUANTITY,
    ...payload
});

export const startDiscountSelectedProduct = payload => ({
    type: START_DISCOUNT_PRODUCT_QUANTITY,
    ...payload
});