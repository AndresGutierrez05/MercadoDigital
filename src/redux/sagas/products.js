import { put, takeLatest } from 'redux-saga/effects';
import { START_GET_PRODUCTS, SUCCESS_GET_PRODUCTS, START_GET_PRODUCT, START_REMOVE_PRODUCT_CART, START_REMOVE_SELECTED_PRODUCT, START_ADD_PRODUCT_QUANTITY, START_DISCOUNT_PRODUCT_QUANTITY, START_SET_CART, 
   SUCCESS_GET_PRODUCT, SUCCESS_SET_CART, SUCCESS_REMOVE_PRODUCT_CART, SUCCESS_REMOVE_SELECTED_PRODUCT, SUCCESS_ADD_PRODUCT_QUANTITY, SUCCESS_DISCOUNT_PRODUCT_QUANTITY  } from '../actions/products';
import productsJSON from "../../data/products.json";

function* getProducts() {
   try {
      yield put({type: SUCCESS_GET_PRODUCTS, payload: productsJSON});
   } catch (e) {
      yield put({type: SUCCESS_GET_PRODUCTS, message: e.message});
   }
}

function* setProductSelected(action){
   try {
      yield put({type: SUCCESS_GET_PRODUCT, payload: action});
   } catch (e) {
      yield put({type: SUCCESS_GET_PRODUCT, message: e.message});
   }
}

function* setCart(action) {
   try {
      yield put({type: SUCCESS_SET_CART, payload: action});
   } catch (e) {
      yield put({type: SUCCESS_SET_CART, message: e.message});
   }
}

function* removeProductCart(action) {
   try {
      yield put({type: SUCCESS_REMOVE_PRODUCT_CART, payload: action});
   } catch (e) {
      yield put({type: SUCCESS_REMOVE_PRODUCT_CART, message: e.message});
   }
}

function* removeSelectedProduct(action) {
   try {
      yield put({type: SUCCESS_REMOVE_SELECTED_PRODUCT, payload: action});
   } catch (e) {
      yield put({type: SUCCESS_REMOVE_SELECTED_PRODUCT, message: e.message});
   }
}

function* addProductQuantity(action) {
   try {
      yield put({type: SUCCESS_ADD_PRODUCT_QUANTITY, payload: action});
   } catch (e) {
      yield put({type: SUCCESS_ADD_PRODUCT_QUANTITY, message: e.message});
   }
}

function* discountProductQuantity(action) {
   try {
      yield put({type: SUCCESS_DISCOUNT_PRODUCT_QUANTITY, payload: action});
   } catch (e) {
      yield put({type: SUCCESS_DISCOUNT_PRODUCT_QUANTITY, message: e.message});
   }
}


// Watchers
export default function* products() {
  yield takeLatest(START_GET_PRODUCTS, getProducts);
  yield takeLatest(START_GET_PRODUCT, setProductSelected);
  yield takeLatest(START_SET_CART, setCart);
  yield takeLatest(START_REMOVE_PRODUCT_CART, removeProductCart);
  yield takeLatest(START_REMOVE_SELECTED_PRODUCT, removeSelectedProduct);
  yield takeLatest(START_ADD_PRODUCT_QUANTITY, addProductQuantity);
  yield takeLatest(START_DISCOUNT_PRODUCT_QUANTITY, discountProductQuantity);
}