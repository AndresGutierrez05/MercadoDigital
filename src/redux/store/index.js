import { createStore, applyMiddleware, combineReducers } from 'redux';
import reduxSaga from 'redux-saga';
import rootSaga from '../sagas';
import productsReducer from "../reducers/productsReducer.js";

const rootReducer = combineReducers({
    productosReducer: productsReducer,
});

const sagaMiddleware = reduxSaga();

export default () => {
    return {
        ...createStore(rootReducer, applyMiddleware(sagaMiddleware)),
        runSaga: sagaMiddleware.run(rootSaga)
    }
};