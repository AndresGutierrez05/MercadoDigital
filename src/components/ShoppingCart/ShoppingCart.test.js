import ShoppingCart from "./ShoppingCart.jsx";
import renderer from "react-test-renderer";
import store from "../../redux/store/index.js";
import { Provider } from "react-redux";
import products from "../../data/products.json";

describe('ShoppingCart component', () => {
    test("default component without product added on cart, totalPrice must to be $0", () => {
        const component = renderer.create(
            <Provider store={store()}>
                <ShoppingCart/>,
            </Provider>
            );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        let testInstance = component.root;
        expect(testInstance.findByProps({ className  : "totalPrice"}).children[1]).toEqual("0");
    });

    if(products !== null && products.length > 0){
        test("component with first product(2) on the cart, totalPrice must to be > $0", () => {
            const component = renderer.create(
                <Provider store={store()}>
                    <ShoppingCart propCart={[{...products[0], cantidad : 2}]}/>,
                </Provider>
                );
    
            
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();

            let testInstance = component.root;
            expect(parseInt(testInstance.findByProps({ className  : "totalPrice"}).children[1])).toBeGreaterThanOrEqual(1);
        });
    }
});