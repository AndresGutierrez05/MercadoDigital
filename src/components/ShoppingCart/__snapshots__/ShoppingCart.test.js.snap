// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart component component with first product(2) on the cart, totalPrice must to be > $0 1`] = `
Array [
  <div
    className="cart-container"
  >
    <div
      className="cart-container-right"
    >
      <div
        className="cart-button"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="cart-button-img"
        >
          <img
            alt="carrito"
            src="/icons/cart.png"
          />
        </div>
        <div>
          <p
            className="totalPrice"
          >
            $
            10800
          </p>
        </div>
      </div>
      <div
        className="cart-summary"
        onMouseLeave={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
      >
        <h2>
          Carrito
        </h2>
        <hr />
        <div
          className="cart-summary-products"
        >
          <div>
            <div
              className="cart-summary-product"
            >
              <div
                className="quantity"
              >
                <div
                  className="quantity-value"
                >
                  <span>
                    2
                  </span>
                </div>
              </div>
              <img
                alt="AtÃºn en Lata"
                src="https://copservir.vtexassets.com/arquivos/ids/773477-800-auto?v=637951138022000000"
              />
            </div>
            <hr />
          </div>
        </div>
        <div
          className="cart-total"
        >
          <div>
            <form
              action="https://checkout.wompi.co/p/"
              method="GET"
            >
              <input
                name="public-key"
                type="hidden"
                value="pub_test_X0zDA9xoKdePzhd8a0x9HAez7HgGO2fH"
              />
              <input
                name="currency"
                type="hidden"
                value="COP"
              />
              <input
                name="amount-in-cents"
                type="hidden"
                value="1080000"
              />
              <input
                name="reference"
                type="hidden"
                value="4XMPGKWWPKWQ"
              />
              <input
                name="signature-integrity"
                type="hidden"
                value="37c8407747e595535433ef8f6a811d853cd943046624a0ec04662b17bbf33bf5"
              />
              <button
                type="submit"
              >
                Paga con 
                <strong>
                  Wompi
                </strong>
              </button>
            </form>
          </div>
          <h3>
            Total : $
            10800
          </h3>
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;

exports[`ShoppingCart component default component without product added on cart, totalPrice must to be $0 1`] = `
Array [
  <div
    className="cart-container"
  >
    <div
      className="cart-container-right"
    >
      <div
        className="cart-button"
        onClick={null}
        onMouseEnter={null}
      >
        <div
          className="cart-button-img"
        >
          <img
            alt="carrito"
            src="/icons/cart.png"
          />
        </div>
        <div>
          <p
            className="totalPrice"
          >
            $
            0
          </p>
        </div>
      </div>
      <div
        className="cart-summary"
        onMouseLeave={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
      >
        <h2>
          Carrito
        </h2>
        <hr />
        <div
          className="cart-summary-products"
        />
        <div
          className="cart-total"
        >
          <div>
            <form
              action="https://checkout.wompi.co/p/"
              method="GET"
            >
              <input
                name="public-key"
                type="hidden"
                value="pub_test_X0zDA9xoKdePzhd8a0x9HAez7HgGO2fH"
              />
              <input
                name="currency"
                type="hidden"
                value="COP"
              />
              <input
                name="amount-in-cents"
                type="hidden"
                value="000"
              />
              <input
                name="reference"
                type="hidden"
                value="4XMPGKWWPKWQ"
              />
              <input
                name="signature-integrity"
                type="hidden"
                value="37c8407747e595535433ef8f6a811d853cd943046624a0ec04662b17bbf33bf5"
              />
              <button
                type="submit"
              >
                Paga con 
                <strong>
                  Wompi
                </strong>
              </button>
            </form>
          </div>
          <h3>
            Total : $
            0
          </h3>
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;
